MANIFEST.in

```
@@ -1,33 +1,4 @@
-graft bootloader
-graft doc
-graft PyInstaller
-graft tests
-
-include *.rst
-include *.txt
-include *.toml
-include pyinstaller.py
-include pyinstaller-gui.py
-
-prune bootloader/build
-prune bootloader/.waf-*
-prune bootloader/.waf3-*
-prune bootloader/waf-*
-prune bootloader/waf3-*
-prune bootloader/_sdks
-prune bootloader/.vagrant
-exclude bootloader/.lock-waf*
-
-prune doc/source
-prune doc/_build
-recursive-exclude doc *.tmp
-include news/_template.rst
-prune news
-
-prune old
-prune scripts
-prune tests/scripts
-prune .github
-
-exclude .* *.yml *~ .directory
-global-exclude *.py[co]
+recursive-include bootloader *.c *.h wscript waf
```

```suggestion
recursive-include bootloader *.c *.h wscript waf strip.py
```

We need to include `bootloader/tools/strip.py`, otherwise trying to build bootloader from sdist fails with:
```
Could not load the Waf tool 'strip' from ['tools',...
```

---

Ughh, in that case then there's some inconsistency somewhere. When I run `python setup.py sdist` that file is already included so it works fine. You're not supposed to have to put Python code in the manifest. This could get rather dangerous if there are other inconsistently missing files lurking in there.

---

Did you try generating `sdist` with completely clean git tree? 

It seems I can get the `tools/strip.py` "accidentally" included if `sdist` is built with "old" `pynstaller.egg-info` present (from previous run that has it on include list); i.e., if I manually remove `dist` but do not remove `pyinstaller.egg-info`, as opposed to doing `git clean`...

---

Ahh that's it. I was keeping a close eye on the `build` folder because I knew that was prone to leaking old files in but I forgot that the `egg-info` could do it too.

---

Guess we just need to run `git clean -xdf` before building the sdist.